source/extensions/transport_sockets/tls/context_impl.cc

```diff
@@ -567,6 +586,66 @@ std::vector<Ssl::PrivateKeyMethodProviderSharedPtr> ContextImpl::getPrivateKeyMe
   return providers;
 }
 
+bool ContextImpl::matchSubjectAltName(
+    X509* cert, const std::vector<Matchers::StringMatcherImpl>& match_subject_alt_name_list) {
+  bssl::UniquePtr<GENERAL_NAMES> san_names(
+      static_cast<GENERAL_NAMES*>(X509_get_ext_d2i(cert, NID_subject_alt_name, nullptr, nullptr)));
+  if (san_names == nullptr) {
+    return false;
+  }
+  for (const GENERAL_NAME* san : san_names.get()) {
```

Comment:
can you extract an utility function to convert GENERAL_NAME to string? then this for will be simpler.